docs/bots-guide.md

```diff
@@ -0,0 +1,192 @@
+* A [documentation](#bot-api) of the bot API.
+* Common [problems](#common-problems) when developing/deploying bots and their solutions.
+
+Contributions to this guide are very welcome, so if you run into any
+issues following these instructions or come up with any tips or tools
+that help writing bots, please visit chat.zulip.org, open an issue, or submit a pull request
+to share your ideas!
+
+## How to deploy a bot
+This guide will show you how to deploy a bot on your running Zulip server. It presumes that you already have a fully implemented `<my-bot>.py` bot and now want to try it out.
+
+1. Copy your bot `<my-bot>.py` to `~/zulip/contrib_bots/lib/`.
+ * This is the place where all Zulip bots are stored.
+ * You can also test out already existing bots in this directory.
+2. Run your Zulip server. Bots can only be deployed on running systems.
+3. Register a new bot at your Zulip server's web interface.
+ * Navigate to *Settings* -> *Your Bots* -> *Add a New Bot*, fill out the form and click on *Create Bot*.
+ * A new bot should appear in the *Your Bots* panel.
+4. Add the bot's configuration file on your Zulip server.
+ * In the *Your Bots* panel, click on the green icon to download its configuration file *.zuliprc* (the structure of this file is explained [here](#configuration-file)
+ * Copy the file to a destination of your choice on your Zulip server, e.g. to `~/.zuliprc`or ~/zuliprc-test.
+5. Subscribe the bot to any streams that the bot needs to read message from or write message to.55
+ * To subscribe your bot to streams, navigate to *Manage Streams*. Select a stream and add your bot by its email address (the address you assigned in step 2).
+ * Now, the bot will do its job on the streams you subscribed it to.
+6. Run the bot.
+ * On your Zulip server, navigate to `~/zulip/contrib_bots/`
+ * Run `python run.py ~/zulip/contrib_bots/<my-bot>.py --config-file ~/.zuliprc
+ * Check the output of the command. It should start with the text the `usage` function returns, followed by logging output similar to this:
+
+>INFO:root:starting message handling...
+>INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): localhost
+
+* Congrats! Now, your bot should be ready to test on the streams you subscribed it to.
+
+### Test the `followup.py` bot
+1. Do the previous steps for the `followup.py` bot.
+2. Subscribe the bot to *followup* and another stream, e.g. *social*.
+3. Send a message to the stream *social* . If everything works, a copy of this message should now pop up in the *followup* stream.
+
+## How to develop a bot
+
+Below explains the structure of a bot `<my-bot>.py`. You can use this as boilerplate code for developing your own bot.
+
+Every bot is built upon this structure:
+```
+class FollowupHandler(object):
+    '''
+    A docstring documenting this bot.
+    '''
+
+    def usage(self):
+        return '''Your description of the bot'''
+
+    def triage_message(self, message):
```

Comment:
Just a heads up that we may add another parameter here called "client."  No need to address this now.